src/test/java/teammates/test/cases/BaseTestCaseWithDatastoreAccess.java

```diff
@@ -187,6 +187,7 @@ private void equalizeIrrelevantData(AccountAttributes expected, AccountAttribute
                 expected.studentProfile.googleId = actual.googleId;
             }
             expected.studentProfile.modifiedDate = actual.studentProfile.modifiedDate;
+            expected.sanitizeForSaving();
```

Comment:
What happens is that the data will be sanitized before saving, and data can be entered in unsanitized form in the json, as it is needed to test proper sanitization after sanitization of html before saving of some attributes is removed. So it is expected that the data should actually be sanitized, and the "expected data" directly from json is actually inaccurate. 
However, in most cases, whether or not the expected data is sanitized is irrelevant to the test cases and it may even be more convenient that the "expected data" contains the original unsanitized data so there the tester knows what to properly assert. So it is much more convenient that `expected` is only sanitized here when necessary, to avoid unnecessary changes to other test cases.